Support pandas 1.0

ajkerrigan opened the pull request:
Upgrade pandas to 1.0 and fix a few pandas issues with vd 2.x. The high level view:

* Support pandas 1.0
* Add JSON Lines support
* Fix a few errors I've bumped into (with some inspiration from the generic loader testing ideas in #507)
* Add an initial set of tests to cover the fixes

ajkerrigan committed:
Upgrade pandas to 1.0.3, pin deps with pip-tools

ajkerrigan committed:
Add .jsonl support for the pandas loader

ajkerrigan committed:
[pandas] Fix selections

* Use `addColumn` rather than setting columns in one shot in a listcomp
* Fix reference to `addUndo`

ajkerrigan committed:
[pandas] Fix row selection/deselection

ajkerrigan committed:
[pandas] First pass at fixing row deletion

Undo is not working properly yet.

saulpw commented:
Cool, @ajkerrigan!  Looking forward to how this shapes up.

ajkerrigan committed:
[cmdlog] find target sheet by name or Sheet object

The `sheet` portion of a command log row sometimes appears as a string
and sometimes as a Sheet object; handle both cases in the replay
context.

ajkerrigan committed:
[pandas] Improve row deletion

Get undo working for individual row deletion, clarify references to the
underlying DataFrame to avoid clashes.

ajkerrigan committed:
[pandas] Improve regex select/unselect logic

Use consistent logic for regex selection/deselection. Make
matches case-insensitive and asynchronous.

ajkerrigan committed:
[pandas] Get undos working properly for row adds/deletes

Make undo work properly for individual or bulk additions and deletions.

anjakefala commented on requirements.in:
```diff
@@ -0,0 +1,24 @@
+python-dateutil     # date type
```
What's this? :)

ajkerrigan replied:
Ah, that's moved over from here as part of pinning:

https://github.com/saulpw/visidata/pull/563/files#diff-b4ef698db8ca845e5845c4618278f29aL1

That whole commit can be left out really, it's too opinionated for this PR. I just figured I was in over my head messing with this stuff and didn't want the extra risk of dependency version changes causing regressions. And it does invite discussion! :)

ajkerrigan committed:
[pandas] Add loader tests

ajkerrigan committed:
Revert "Upgrade pandas to 1.0.3, pin deps with pip-tools"

This (mostly) reverts commit 59ef39dcb33db17be7cbf7323b189df6c35b55cc.

The only change we keep is pandas 1.0.3

saulpw commented on visidata/loaders/_pandas.py:
```diff
@@ -173,13 +215,101 @@ def _selectByILoc(self, mask, selected=True):
+        '''
+        import pandas as pd
+        masks = pd.DataFrame([
+            self.df[col.name].astype(str).str.contains(pat=regex, case=False, regex=True)
```
`case=('I' not in options.regex_flags)` I think.

ajkerrigan replied:
Good call 👍

anjakefala commented on visidata/loaders/_pandas.py:
```diff
@@ -55,42 +61,77 @@ def read_tsv(self, path, **kwargs):
+
+        self.columns = []
+        for col in (c for c in df.columns if not c.startswith("__vd_")):
+            self.addColumn(Column(
```
++ for `self.addColumn` =D

ajkerrigan committed:
Allow vd.getSheet to handle Sheet instances

When looking up a Sheet instance from vd.getSheet(), return the instance
as-is. This helps keep lookup logic in vd.moveToReplayContext() simpler.

This reverts commit 0deddbe15a50ce04dfdc3beb041a9d3af1e1987d.

ajkerrigan committed:
[pandas] Add undo to tests

ajkerrigan committed:
[pandas] Take regex case sensitivity from regex_flags option

anjakefala commented:
Thanks for doing all the requested changes @ajkerrigan! =) And for doing all of this wonderful work for the pandas loader. :purple_heart: :purple_heart: I am going to go ahead and merge it in.
